Task7.py

```diff
@@ -0,0 +1,35 @@
+""" 7 Шоколадка **
+Шоколадка имеет вид прямоугольника, разделенного на n×m долек.
+ Шоколадку можно один раз разломить по прямой на две части.
+Определите, можно ли таким образом отломить от шоколадки ровно k долек.
+Определите, можно ли отломить от шоколадки ровно k одиночных долек
+Определите, можно ли отломить от шоколадки ровно k одиночных долек
+с помощью m разломов"""
+
+n = int(input())
+m = int(input())
+k = int(input())
+
+if (n % 2 == 0 or m % 2 == 0) and (n * m) % k == 0:
+    """ 1) проверяем условие (шоколадку один раз можно
+    разломить по прямой на две части)
+    2) задаем условие при которм при делении на
+    ровно K долек не останеьтся остатка """
+    print('можно разоломить на ровно k долек')
+elif (n % 2 == 0 or m % 2 == 0) and (n * m) / k == 1:
+    """ 1) проверяем условие (шоколадку один раз можно
+    разломить по прямой на две части)
+    2) задаем условие при которм при делении на
+    K долек не останеьтся остатка (фактически мы всю
+    шоколадку ламая по полам разламаем на дольки"""
+    print('можно разоломить на ровно k одиночных долек')
+elif (n % 2 == 0 or m % 2 == 0) and k == ((m - 1) * (n - 1)):
+    """ 1) проверяем условие (шоколадку один раз можно
+    разломить по прямой на две части)
+        2) чтобы разделить шоколадку на m разломов
+    эксперементальным путем установлено, что k должна
+    соответствовать условию ((m - 1) * (n - 1))"""
```

Comment:
there are a lot of  other options. i.e. on a chocolate 100(n)x10(m) it would be possible to cut 4(k) 1x1 squares with 10 (m) cuts. 
Additionally, I've corrected task (k squares 1x1 for t cuts)

Reply:
I will still think about a solution.